src/wwwroot/app/directives/event-calendar/event-calendar.js

```diff
@@ -33,59 +35,82 @@ function EventCalendarController($scope) {
    vm.onDblclick          = showAddDialog;
    vm.eventsForMonth      = {};
    vm.todayDate           = new Date();
+   vm.selectedDay         = new Date();
    vm.dateForSearchEvents = new Date();
    vm.currentMonth        = vm.dateForSearchEvents.getMonth();
    vm.currentYear         = vm.dateForSearchEvents.getFullYear();
    vm.goToNextMonth       = goToNextMonth;
    vm.goToPreviousMonth   = goToPreviousMonth;
+   vm.goToNextYear        = goToNextYear;
+   vm.goToPreviousYear    = goToPreviousYear;
    vm.removeCurrentEvent  = removeCurrentEvent;
    vm.saveCurrentEvent    = saveCurrentEvent;
    vm.onDblclick          = showAddDialog;
    vm.checkedUsers        = [];
+   vm.eventsForDay        = [];
+   vm.getEventsForDate    = getEventsForDate;
+   vm.daysOfWeek          = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
 
-   function init() {
-      getDaysInMonth(vm.currentMonth, vm.currentYear);
-      vm.getEvents(new Date(), vm.chekedUsers).then((events) => {
+   (function init() {
+      formatingSearchCondition();
+      getDaysInMonth(vm.startDate, vm.endDate);
+      vm.getEvents(vm.startDate, vm.endDate, vm.checkedUsers).then((events) => {
```

Comment:
you can call method in 'then' instead of creating a new clause. like 
**vm.getEvents(...).then(convertEventsToHash);**